### PR TITLE
Add 2 extra Confirmation Modals in order to delete an account

### LINKED ISSUE #2583: [FEATURE] Delete Acccount 3rd confirmation
### Please confirm the following

- [X] I checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate issues

### What do you want to be added or improved?

As a user, I want to not delete my accounts by accident/mistake. That's why I need a very serious final confirmation before deleting an account.

### Why do you need it?

- Users accidentally delete accounts and lose data
- People don't read the delete confirmation and just click delete

### How do you imagine it?

After confirming delete a 3rd delete confirmation must show up.
In this final delete confirmation, the user must type the name of the Account correctly in order to delete.
You can see GitHub has a similar UX when deleting repos.

You can use a Material3 dialog with TextField inside it.

UX goal:
- User clicks delete account
- A confirmation modal shows up (already implemented)
- User clicks delete again
- A final confirmation dialog shows.
- The user must type the name of the account.
- After typing the name of the account, the final "DELETE PERMANENTLY" button will become enabled.
- After the user clicks the final "DELETE PERMANENTLY" button thee account will be gone for good.

### PR DESCRIPTION
**Pull request by mohammednawas8**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [X] The PR is submitted to the `main` branch.
- [X] I've read the **[Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md)**.
- [X] The code builds and is tested on a real Android device.
- [X] I confirm that I've run the code locally and everything works as expected.

## What's changed?

Describe with a few bullets **what's new:**
- a delete account 2nd Confirmation Modal
- b delete account 3rd Confirmation Modal

_💡 Tip: Please, attach screenshots and screen recordings. It helps a lot!_

## Risk Factors

**What may go wrong if we merge your PR?**

- a
- b
- c

**In what cases your code won't work?**

- a
- b
- c

## Does this PR closes any GitHub Issues?

Check **[Ivy Wallet Issues](https://github.com/Ivy-Apps/ivy-wallet/issues)**.

- Closes #2583 

_Replace `{ISSUE_NUMBER}` with the id/number of the issue that you've fixed._

### CODE REVIEW

**Comment by ILIYANGERMANOV** [comment]
Hey, please ignore the CI failures - it's not you. Missing GitHub secrets permissions most likely

**Comment by mohammednawas8** [comment]
Alright then

**Comment by ILIYANGERMANOV** [comment]
# Lint

I see missing translations failure for Lint:
```
 Fix the issues identified by lint, or create a baseline to see only new errors.
For more on this, please refer to https://docs.gradle.org/8.3-rc-2/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
  To create a baseline, run `gradlew updateLintBaseline` after adding the following to the module's build.gradle file:
179 actionable tasks: 179 executed
  ```
  android {
      lint {
          baseline = file("lint-baseline.xml")
      }
  }
  ```
  For more details, see https://developer.android.com/studio/write/lint#snapshot
  
  Lint found 2 errors, 116 warnings. First failure:
  
  /home/runner/work/ivy-wallet/ivy-wallet/ivy-resources/src/main/res/values/strings.xml:132: Error: "account_confirm_deletion_type_account_name" is not translated in "ar" (Arabic), "hi" (Hindi), "mn" (Mongolian), "kn" (Kannada), "ru" (Russian), "bg" (Bulgarian), "it" (Italian), "es" (Spanish) [MissingTranslation]
      <string name="account_confirm_deletion_type_account_name">Please type \"%s\" in order to delete your account.</string>
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  
  The full lint text report is located at:
    /home/runner/work/ivy-wallet/ivy-wallet/ivy-resources/build/intermediates/lint_intermediate_text_report/release/lint-results-release.txt
```

You can add a lint baseline to fix it

# Detekt

Detekt seems to found some valid formatting, missing imports warnings. Check them out in the work run.


# Fix incoming

I just merged https://github.com/Ivy-Apps/ivy-wallet/pull/2608 which should solve the CI secret problem for you. I'll update your branch

File: app/src/main/java/com/ivy/wallet/ui/statistic/level2/ItemStatisticScreen.kt [hunk 1]
```diff
@@ -158,6 +164,11 @@ fun BoxWithConstraintsScope.ItemStatisticScreen(screen: ItemStatistic) {
         account = account,
         category = category,
 
+
+        accountNameConfirmation = accountNameConfirmation,
+        updateAccountNameConfirmation = { viewModel.updateAccountDeletionState(it) },
```
Comment (ILIYANGERMANOV) [inline app/src/main/java/com/ivy/wallet/ui/statistic/level2/ItemStatisticScreen.kt:169]: nit: I guess you use:
`        updateAccountNameConfirmation = viewModel::updateAccountDeletionState,`

**Review by ILIYANGERMANOV — APPROVED** [review]
LGTM! Great work and nice UX 💯 Just fix the Lint + Detekt CI problems and we're good to go. Please, tag me if Detekt reports false-positives or something stupid

**Comment by ILIYANGERMANOV** [comment]
# UPDATE

Hey @mohammednawas8  I analyzed the Detekt report. You can ignore the **"complexity: 3"** errors. They're because of the Ivy Wallet legacy code - it's not you.

1) Fix the rest
2) After you've fixed them run: `./gradlew detektBaseline`
3) Ping me for re-review 👍 

Apologies for the flaky CI - we're the process of improving our dev experience and configuring stuff to make contributing more fun

**Comment by mohammednawas8** [comment]
got it

**Comment by ILIYANGERMANOV** [comment]
Hey @mohammednawas8 just a heads up, you might need to rebase your feature and fix merge conflicts cuz I'll merge a huge PR for modularizing Ivy Wallet: https://github.com/Ivy-Apps/ivy-wallet/pull/2611

The goal is to split Ivy Wallet into logically divided modules instead of having a big `:app` monolith.

**Comment by mohammednawas8** [comment]
Alright sure, just note that i am not working on this issue lately, but i will continue working on it after the merge.

**Comment by ILIYANGERMANOV** [comment]
no worries^ no rush 😅

**Comment by mohammednawas8** [comment]
Hi, so i have tried to add a lint baseline to ignore the translation but when i run the lint check it actually keeps detecting that there is a translation missing.
I suggest to use Google translate to just translate that string to the other languages.

**Comment by ILIYANGERMANOV** [comment]
Hey @mohammednawas8 just rebase your PR, I have disabled the "MissingTranslation" check because it's bad dwv experience

**Comment by ILIYANGERMANOV** [comment]
Just fyi: something is wrong with the rebase/merge, **Files changed: 551**. I assume that you've rebased on an older version of `main`. If you pull the latest `main` and base your PR on top of it, most of the CI checks should pass w/o changes ✅

**Comment by mohammednawas8** [comment]
Hi, i just created another branch for this issue, at the begging Lint showed "MissingTranslation" error again, but i ended it up by adding the translatable="false" attribute and that worked. 
i will create a new pull request for the changes.

**Comment by ILIYANGERMANOV** [comment]
> Hi, i just created another branch for this issue, at the begging Lint showed "MissingTranslation" error again, but i ended it up by adding the translatable="false" attribute and that worked. i will create a new pull request for the changes.

Apologies for the nad experience! translatable = false is perfectly for 💯 I tried to disable this Lint check but it seems that it didn't work

**Comment by ILIYANGERMANOV** [comment]
Closed in favor of https://github.com/Ivy-Apps/ivy-wallet/pull/2641

**Pull request closed without merging**